build: Mark print-% target as phony.

dgoncharov opened the pull request:
.PHONY does not take patterns (such as print-%) as prerequisites.
Have print-% depend on force and mark force as phony.

This change ensures print-% rule works even when there is a file that matches the target.

```
$ # on master
$ make print-host
host=x86_64-pc-linux-gnu
$ touch print-host
$ make print-host
make: 'print-host' is up to date.
$
$ git co mark_print_as_phony
Switched to branch 'mark_print_as_phony'
$ make print-host
host=x86_64-pc-linux-gnu
$ touch force
$ make print-host
host=x86_64-pc-linux-gnu
```

gtkiller committed:
Mark print-% target as phony.

.PHONY does not take patterns (such as print-%) as prerequisites.
Have print-% depend on FORCE and mark FORCE as phony.

$ # on master
$ make print-host
host=x86_64-pc-linux-gnu
$ touch print-host
$ make print-host
make: 'print-host' is up to date.
$
$ git co mark_print_as_phony
Switched to branch 'mark_print_as_phony'
$ make print-host
host=x86_64-pc-linux-gnu
$ touch FORCE
$ make print-host
host=x86_64-pc-linux-gnu

hebasto commented:
Also `Makefile.am` and `src/Makefile.am`?

dgoncharov commented:
Sure.

dongcarl commented:
Nice! Wondering why there's a difference in how you do it for `Makefile`s and `Makefile.am`s

dgoncharov commented:
What is the difference?

dongcarl commented:
> What is the difference?

1. It seems like you capitalize `FORCE` in `Makefile.am` and `src/Makefile.am`, whereas it is `force` in `depends/Makefile`
2. You do `.PHONY: force` in `depends/Makfile`, but not in `Makefile.am` or `src/Makefile.am`

dgoncharov commented:
Both of these differences are accidental.
i initially planned to only modify depends/Makefile. So, i added target 'force'.
Then Hennadii pointed out that Makefile.am and src/Makefile.am have the same print-% rule.
So, i modified Makefile.am and src/Makefile.am. Both Makefile.am and src/Makefile.am already had a phony target called 'FORCE'. This eliminated the need for 'force' and i made print-% depend on 'FORCE' in Makefile.am and src/Makefile.am.

dongcarl commented:
Ah right! Let's just capitalize the one in `depends/Makefile` to make them consistent.

Code Review ACK a27d4a3f357dde2a0f2b27794c716160846d78fe modulo `FORCE` capitalization in `depends/Makefile`

dongcarl commented:
Code Review ACK 020802c5b06489190f48bf544de761c72e2cce43

-----

We might need to squash the commits before merge, but everything else looks good

fanquake commented:
Yes this needs squashing.

hebasto reviewed (approved):
ACK fb7be92b094477131140b58a4e3ae98366b93e76, tested on Linux Mint 20.2 (x86_64).
